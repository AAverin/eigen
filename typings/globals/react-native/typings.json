{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
