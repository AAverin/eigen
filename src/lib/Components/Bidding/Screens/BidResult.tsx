import React from "react"
import styled from "styled-components/native"

import { View } from "react-native"
import { Timer } from "../Components/Timer"
import { Title } from "../Components/Title"

interface BidResultProps {
  winning: boolean
}

export class BidResult extends React.Component<BidResultProps> {
  render() {
    return (
      <CenteringContainer>
        <View>
          <TopOffset>
            <Icon source={require("../../../../../images/circle-check-green.png")} />
            <Title>You're the highest bidder</Title>

            <TimeLeft>Time left</TimeLeft>
            <Timer timeLeftInMilliseconds={1000 * 60 * 20} />
          </TopOffset>
        </View>
      </CenteringContainer>
    )
  }
}

const TopOffset = styled.View`
  top: -50%;
  align-items: center;
`

const CenteringContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`

const Icon = styled.Image`
  width: 20px;
  height: 20px;
  margin: 10px;
`

const TimeLeft = styled.Text`
  font-size: 12px;
  color: #666666;
  margin-top: 10px;
`
