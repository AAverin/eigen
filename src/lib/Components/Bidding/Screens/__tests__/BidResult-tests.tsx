import React from "react"
import "react-native"

import * as renderer from "react-test-renderer"
import { BidResult } from "../BidResult"

const saleArtwork = {
  current_bid: {
    amount: "CHF10,000",
    cents: 1000000,
    display: "CHF 10,000",
  },
  sale: {
    live_start_at: "2022-01-01T00:03:00+00:00",
    end_at: "2022-05-01T00:03:00+00:00",
  },
}
describe("BidResult component", () => {
  Date.now = jest.fn(() => 1525983752116)
  it("renders winning screen properly", () => {
    jest.useFakeTimers()

    const bg = renderer.create(<BidResult winning sale_artwork={saleArtwork} navigator={jest.fn() as any} />).toJSON()

    expect(bg).toMatchSnapshot()
    expect(setInterval).toHaveBeenCalledTimes(1)
  })

  it("renders not highest bid screen properly", () => {
    jest.useFakeTimers()
    const messageHeader = "Your bid wasn’t high enough"
    const messageDescriptionMd = `Hello Hello [Your](http://example.com) bid didn’t meet the reserve price for this work.

  Bid again to take the lead.`
    const bg = renderer
      .create(
        <BidResult
          winning={false}
          sale_artwork={saleArtwork}
          message_header={messageHeader}
          message_description_md={messageDescriptionMd}
          navigator={jest.fn() as any}
        />
      )
      .toJSON()
    expect(bg).toMatchSnapshot()
    expect(setInterval).toHaveBeenCalledTimes(1)
  })
})
