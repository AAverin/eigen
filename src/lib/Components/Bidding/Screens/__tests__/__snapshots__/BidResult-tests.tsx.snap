// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders winning screen properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "top": "-50%",
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/circle-check-green.png",
          }
        }
        style={
          Array [
            Object {
              "height": 20,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 20,
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "AGaramondPro-Regular",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        You're the highest bidder
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#666666",
              "fontSize": 12,
              "marginTop": 10,
            },
            undefined,
          ]
        }
      >
        Time left
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        00
        d 
        00
        h
         
        20
        m 
        00
        s
      </Text>
    </View>
  </View>
</View>
`;
